web/src/views_util.js

```diff
@@ -25,7 +27,9 @@ export function show(opts) {
 
     // Hide selected elements in the left sidebar.
     opts.highlight_view_in_left_sidebar();
-    stream_list.handle_message_view_deactivated();
+    if (!opts.recent_view_stream_id) {
+        stream_list.handle_message_view_deactivated();
+    }
```

Comment:
This should probably be unconditional; we would call a second function to show the right thing when loading the per-stream recent view.

Reply:
Yeah that happens in `opts.highlight_view_in_left_sidebar();` which is right above this line, I can just swap them.